crates/shared/src/lib.rs

```diff
@@ -57,19 +57,10 @@ pub type Web3Transport = DynTransport;
 pub type Web3 = DynWeb3;
 pub type Web3CallBatch = CallBatch<Web3Transport>;
 
-/// The standard http client we use in the api and driver.
-pub fn http_client(timeout: Duration) -> reqwest::Client {
-    reqwest::ClientBuilder::new()
-        .timeout(timeout)
-        .user_agent("cowprotocol-services/2.0.0")
-        .build()
-        .unwrap()
-}
-
 /// Create a Web3 instance.
-pub fn web3(client: &Client, url: &Url, name: impl ToString) -> Web3 {
+pub fn web3(http_factory: &HttpClientFactory, url: &Url, name: impl ToString) -> Web3 {
     let transport = Web3Transport::new(HttpTransport::new(
-        client.clone(),
+        http_factory.configure(|builder| builder.cookie_store(true)),
```

Comment:
In particular here, we enable a cookie store so that we can store session cookies for our load-balancer "sticky sessions".

Reply:
cc @ahhda and @fedgiac